sd-rsyslog

```diff
@@ -74,15 +74,33 @@ def onInit():
     # emitted you must set 'level' to logging.DEBUG above.)
     logging.debug("onInit called")
 
-
     global process
     if not os.path.exists("/etc/sd-rsyslog.conf"):
         print("Please create the configuration file at /etc/sd-rsyslog.conf", file=sys.stderr)
         sys.exit(1)
     config = configparser.ConfigParser()
     config.read('/etc/sd-rsyslog.conf')
     logvmname = config['sd-rsyslog']['remotevm']
-    localvmname = config['sd-rsyslog'].get('localvm', gethostname())
+    localvmname = config['sd-rsyslog'].get('localvm', None)
+
+    # If no localvm name is specified, it must be supplied by Qubes OS. If this
+    # fails, we exit, to avoid falsely identified logs.
+    if localvmname is None:
+        try:
+            get_vm_name_process = Popen(["/usr/bin/qubesdb-read", "/name"],
```

Comment:
Rather than `Popen`, we could use `subprocess.check_output()` and catch `CalledProcessError` along with `FileNotFoundError` below, which would avoid the need to inspect the stderr. Not blocking, though, both are easy enough to read.